### PR TITLE
TOOLS-3574 Run tests against 8.0 and add ubuntu2404 variants

### PR DESCRIPTION
**Pull request by jddubois**
Runs integration tests against MongoDB 8.0 and adds dist, sign, and push tasks for ubuntu2404 variants.

### CODE REVIEW

File: release/platform/platform.go [hunk 1]
```diff
@@ -536,7 +536,7 @@ var platforms = []Platform{
 		OS:        OSLinux,
 		Pkg:       PkgDeb,
 		Repos:     []Repo{RepoEnterprise, RepoOrg},
-		BuildTags: []string{"failpoints"},
+		BuildTags: defaultBuildTags,
```
Comment (jddubois) [inline release/platform/platform.go:539]: This works as is, I don't think there's any reason we need to not support Kerberos on ubuntu arm variants.

Comment (autarch) [reply release/platform/platform.go:539]: I'm pretty sure this wasn't done intentionally when these platforms were added. We probably should've caught it in PR review. Good to fix it now!

**Review by autarch — APPROVED** [review]
LGTM!

**Review by tdq45gj — COMMENTED** [review]
Do we need to push to the Ubuntu2404 server repo? If so, I think we should add the new platforms to `etc/repo-config.yml` as well.

**Comment by jddubois** [comment]
> repo-config.yml

Great catch- thanks! Updated that with codename `noble`, which I see some references to in Slack so I think that's correct.

**Review by tdq45gj — APPROVED** [review]
LGTM!

**Pull request merged**